File: 2114/python/2114.py
# python 2114.py

class Solution(object):
    def mostWordsFound(self, sentences):
        """
        :type sentences: List[str]
        :rtype: int
        """
        maxWords = 0

        for sentence in sentences:
            maxWords = len(sentence.split(" ")) if (
                maxWords < len(sentence.split(" "))) else maxWords

        return maxWords

solution = Solution()

print(solution.mostWordsFound(
    ["alice and bob love leetcode", "i think so too", "this is great thanks very much"]))
print(solution.mostWordsFound(
    ["please wait", "continue to fight", "continue to win"]))

File: 2114/c#/2114.cs
// mcs -out:2114.exe 2114.cs
// mono 2114.exe

using System;

public class Solution
{
    public int MostWordsFound(string[] sentences)
    {
        int maxWords = 0;

        foreach (var sentence in sentences)
        {
            maxWords =
                (maxWords < sentence.Split(" ").Length)
                    ? sentence.Split(" ").Length
                    : maxWords;
        }

        return maxWords;
    }

    static void Main(string[] args)
    {
        Solution solution = new Solution();
        Console
            .WriteLine("[{0}]",
            string
                .Join(", ",
                solution
                    .MostWordsFound(new string[] {
                        "alice and bob love leetcode",
                        "i think so too",
                        "this is great thanks very much"
                    })));

        Console
            .WriteLine("[{0}]",
            string
                .Join(", ",
                solution
                    .MostWordsFound(new string[] {
                        "please wait",
                        "continue to fight",
                        "continue to win"
                    })));
    }
}
